corehq/apps/hqwebapp/management/commands/synch_phone_nums.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.core.management.base import BaseCommand
+
+from corehq.apps.users.models import CommCareUser
+from corehq.apps.sms.tasks import sync_user_phone_numbers as sms_sync_user_phone_numbers
+from corehq.form_processor.models import CommCareCaseSQL
+from corehq.messaging.tasks import sync_case_for_messaging
+from corehq.util.log import with_progress_bar
+from corehq.sql_db.util import get_db_aliases_for_partitioned_query
+
+
+class Command(BaseCommand):
+    help = "Resync all contacts' phone numbers for given projects"
+
+    def add_arguments(self, parser):
+        parser.add_argument('-p', '--projects', type=str, help='Comma separated project list')
+
+    def handle(self,  **kwargs):
+        projects = kwargs['projects'].split(',')
+        db_aliases = get_db_aliases_for_partitioned_query()
+        db_aliases.sort()
+
+        for domain in projects:
+            print("Resync all contacts' phone numbers for project %s  " % domain)
+            print("Synching for phone numbers")
+            commcare_user_ids = (
+                    CommCareUser.ids_by_domain(domain, is_active=True) +
+                    CommCareUser.ids_by_domain(domain, is_active=False)
+            )
+            for user_id in with_progress_bar(commcare_user_ids):
+                sms_sync_user_phone_numbers.delay(user_id)
+
+            print("Iterating over databases: %s" % db_aliases)
+            for db_alias in db_aliases:
+                print("")
+                print("Synching for cases in %s ..." % db_alias)
+                case_ids = list(
+                    CommCareCaseSQL
```

Comment:
This will only return cases from SQL domains, but there are still a fair amount of domains that store cases in couch. This should be changed to query the proper backend as per [`should_use_sql_backend`](https://github.com/dimagi/commcare-hq/blob/33158b9118200867a3f14fc25df1695f2a0f531f/corehq/form_processor/utils/general.py#L31 ).

Reply:
Just one questions how come some cases are stored in sql and some in couch

Reply:
Originally HQ only used couch for data storage. More recently we began moving forms and cases to postgresql for scale and performance reasons. We're currently in the progress of moving all forms and cases to SQL, but that process is not finished.

Reply:
cool. That means ABT project should also be pretty old.